php-classes/Slate/CBL/Tasks/StudentTasksRequestHandler.php

```diff
@@ -60,6 +61,32 @@ protected static function buildBrowseConditions(array $conditions = [], array &$
             ];
         }
 
+        // apply disabled students filter
+        if (!static::getRequestedStudentDisabledFilter()) {
+            $disabledStudentIds = DB::allValues(
```

Comment:
I don't looovee this approach of getting a list of IDs first to exclude, but after thinking about it a bit I do think it's our best option, and this is a pattern we use in a lot of places.

What I'd like to do though is create a public static method on [Slate's override of the `Person` model class](https://github.com/SlateFoundation/slate/blob/develop/php-classes/Emergence/People/Person.php) like `Person::getDeactivatedIds()`

Then that method can cache the ID list like [this method does](https://github.com/SlateFoundation/slate-cbl/blob/develop/php-classes/Slate/CBL/Competency.php#L121)—except in this case we don't need to include any ID in the cache key, it could be a static global key like `deactivated-people-ids`. And then you can have an after-save handler like this that just blows out that whole key any time an `AccountLevel` field changes [like this method does](https://github.com/JarvusInnovations/emergence-skeleton/blob/develop/php-classes/Emergence/CMS/ContentBlock.php#L76)

The list of deactivated account IDs won't change frequently and we'll want to use it in a lot of filters so I think it makes sense to let the `Person` model handle caching a static ID list for us